### ISSUE TITLE
Running the server container without internet

### ISSUE THREAD

**Issue by cglewis** [rancher/rancher#924]
Is there a way to have the server container pull down whatever it needs from `releases.rancher.com` on build, rather than every time it's started?

```
2015-05-26 16:07:29,704 ERROR [:] [] [] [] [main           ] [i.c.p.i.a.s.filter.UIPathFilter     ] Failed to load UI from [http://releases.rancher.com/ui/0.20.3/static/index.html] java.net.UnknownHostException: releases.rancher.com
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:178) ~[na:1.7.0_79]
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[na:1.7.0_79]
	at java.net.Socket.connect(Socket.java:579) ~[na:1.7.0_79]
	at java.net.Socket.connect(Socket.java:528) ~[na:1.7.0_79]
	at sun.net.NetworkClient.doConnect(NetworkClient.java:180) ~[na:1.7.0_79]
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:432) ~[na:1.7.0_79]
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:527) ~[na:1.7.0_79]
	at sun.net.www.http.HttpClient.<init>(HttpClient.java:211) ~[na:1.7.0_79]
	at sun.net.www.http.HttpClient.New(HttpClient.java:308) ~[na:1.7.0_79]
	at sun.net.www.http.HttpClient.New(HttpClient.java:326) ~[na:1.7.0_79]
	at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:997) ~[na:1.7.0_79]
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:933) ~[na:1.7.0_79]
	at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:851) ~[na:1.7.0_79]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1301) ~[na:1.7.0_79]
	at io.cattle.platform.iaas.api.servlet.filter.UIPathFilter.reloadIndex(UIPathFilter.java:57) [cattle-iaas-api-logic-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.iaas.api.servlet.filter.UIPathFilter.init(UIPathFilter.java:35) [cattle-iaas-api-logic-0.5.0-SNAPSHOT.jar:na]
	at org.eclipse.jetty.servlet.FilterHolder.doStart(FilterHolder.java:119) [jetty-servlet-8.1.11.v20130520.jar:8.1.11.v20130520]
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:64) [jetty-util-8.1.11.v20130520.jar:8.1.11.v20130520]
	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:719) [jetty-servlet-8.1.11.v20130520.jar:8.1.11.v20130520]
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:265) [jetty-servlet-8.1.11.v20130520.jar:8.1.11.v20130520]
	at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1252) [jetty-webapp-8.1.11.v20130520.jar:8.1.11.v20130520]
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:710) [jetty-server-8.1.11.v20130520.jar:8.1.11.v20130520]
	at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:494) [jetty-webapp-8.1.11.v20130520.jar:8.1.11.v20130520]
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:64) [jetty-util-8.1.11.v20130520.jar:8.1.11.v20130520]
	at org.eclipse.jetty.server.handler.HandlerWrapper.doStart(HandlerWrapper.java:95) [jetty-server-8.1.11.v20130520.jar:8.1.11.v20130520]
	at org.eclipse.jetty.server.Server.doStart(Server.java:282) [jetty-server-8.1.11.v20130520.jar:8.1.11.v20130520]
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:64) [jetty-util-8.1.11.v20130520.jar:8.1.11.v20130520]
	at io.cattle.platform.launcher.jetty.Main.main(Main.java:125) [0.5.0-SNAPSHOT-75f3d3ca72ef2db89c04461a50dfca191d649a6d-0b8b0f97-9926-4101-b226-ccfa9077f956/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_79]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_79]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_79]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_79]
	at io.cattle.platform.launcher.Main.run(Main.java:188) [0.5.0-SNAPSHOT-75f3d3ca72ef2db89c04461a50dfca191d649a6d-0b8b0f97-9926-4101-b226-ccfa9077f956/:na]
	at io.cattle.platform.launcher.Main.main(Main.java:251) [0.5.0-SNAPSHOT-75f3d3ca72ef2db89c04461a50dfca191d649a6d-0b8b0f97-9926-4101-b226-ccfa9077f956/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_79]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_79]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_79]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_79]
	at io.cattle.platform.packaging.Bootstrap.run(Bootstrap.java:383) [cattle.jar:0.5.0-SNAPSHOT]
	at io.cattle.platform.packaging.Bootstrap.main(Bootstrap.java:424) [cattle.jar:0.5.0-SNAPSHOT]
```

This was a previously working container, but after the host running it got shut down and it came back up it wasn't able to resolve which forced me to create a new server container and then create all new agent containers for all of my hosts.  It seems like it could be a flag or something, if I really want the latest JAR on restart of the container, or if I just want what was available when I first created/pulled down the container.

FWIW.

**Comment by clemenko** [comment]
How much $$$ to make this portable to air-gapped networks?

**Comment by vincent99** [comment]
![image](https://cloud.githubusercontent.com/assets/753917/7948157/c96c5ef4-0936-11e5-878f-81b61d2c36da.png)

..but seriously, the next release has the UI's `static/index.html` baked in so that it doesn't need to hit `releases.rancher.com` to get it.  The UI still needs access to the internet, but hopefully internet access from a browser is less of a problem for you.  And the server does still need to be able to pull images.  But we're getting closer.

**Comment by clemenko** [comment]
I am hoping I will be able to "save" the server and agent images and move it to an air-gapped network. Meaning ZERO internet access.

**Comment by aruneli** [comment]
Verified this using private registry and no internet access

**Comment by clemenko** [comment]
What version of server will work offline?

**Comment by aruneli** [comment]
@clemenko We are going to release v0.34 today or tomorrow. Starting that release, if you have private registry setup, it should work. Expect a detailed document from @deniseschannon along with the release. But there is a caveat - UI still needs access to internet (releases.rancher.com to be specific). But we are working on removing that restriction and it should be fixed in future release.

**Comment by clemenko** [comment]
When do you think that restriction will be removed? I need a complete OFFLINE install.

**Comment by aruneli** [comment]
@clemenko This will be fixed in our next release (v0.35). For now, you will need a machine that has access to both releases.rancher.com and to private air gapped network for UI to work.

**Comment by clemenko** [comment]
Ok. I need to wait then.
